x-pack/plugins/security_solution/cypress/integration/detection_rules/bulk_edit_rules.spec.ts

```diff
@@ -0,0 +1,220 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  ELASTIC_RULES_BTN,
+  CUSTOM_RULES_BTN,
+  MODAL_CONFIRMATION_BTN,
+  SELECT_ALL_RULES_ON_PAGE_CHECKBOX,
+  LOAD_PREBUILT_RULES_ON_PAGE_HEADER_BTN,
+  RULES_TAGS_FILTER_BTN,
+} from '../../screens/alerts_detection_rules';
+
+import {
+  RULES_BULK_EDIT_OVERWRITE_INDEX_PATTERNS_CHECKBOX,
+  RULES_BULK_EDIT_OVERWRITE_TAGS_CHECKBOX,
+  RULES_BULK_EDIT_INDEX_PATTERNS_WARNING,
+  RULES_BULK_EDIT_TAGS_WARNING,
+} from '../../screens/rules_bulk_edit';
+
+import {
+  changeRowsPerPageTo,
+  waitForRulesTableToBeLoaded,
+  selectAllRules,
+  goToTheRuleDetailsOf,
+  waitForRulesTableToBeRefreshed,
+  selectNumberOfRules,
+  testAllTagsBadges,
+} from '../../tasks/alerts_detection_rules';
+
+import {
+  openBulkEditAddIndexPatternsForm,
+  openBulkEditDeleteIndexPatternsForm,
+  typeIndexPatterns,
+  waitForBulkEditActionToFinish,
+  confirmBulkEditForm,
+  clickAddIndexPatternsMenuItem,
+  waitForElasticRulesBulkEditModal,
+  waitForMixedRulesBulkEditModal,
+  openBulkEditAddTagsForm,
+  openBulkEditDeleteTagsForm,
+  typeTags,
+} from '../../tasks/rules_bulk_edit';
+
+import { hasIndexPatterns } from '../../tasks/rule_details';
+import { loginAndWaitForPageWithoutDateRange } from '../../tasks/login';
+
+import { SECURITY_DETECTIONS_RULES_URL } from '../../urls/navigation';
+import { createCustomRule } from '../../tasks/api_calls/rules';
+import { cleanKibana } from '../../tasks/common';
+import {
+  getExistingRule,
+  getNewOverrideRule,
+  getNewRule,
+  getNewThresholdRule,
+  totalNumberOfPrebuiltRules,
+} from '../../objects/rule';
+
+const RULE_NAME = 'Custom rule for bulk actions';
+
+const CUSTOM_INDEX_PATTERN_1 = 'custom-cypress-test-*';
+const DEFAULT_INDEX_PATTERNS = ['index-1-*', 'index-2-*'];
+const TAGS = ['cypress-tag-1', 'cypress-tag-2'];
+const OVERWRITE_INDEX_PATTERNS = ['overwrite-index-1-*', 'overwrite-index-2-*'];
+
+const customRule = {
+  ...getNewRule(),
+  index: DEFAULT_INDEX_PATTERNS,
+  name: RULE_NAME,
+};
+
+describe('Detection rules, bulk edit', () => {
+  beforeEach(() => {
+    cleanKibana();
```

Comment:
This method is really expensive in execution time, correct me if I'm wrong, but I think not generating alerts in those tests or modifying existing indexes. In this case I would suggest in order to save execution time to call it just once in order to make sure the state is clean from previous tests, and create an afterEach hook where we delete the rules we have created. You can take advantage of the `deleteCustomRule` method we have on `x-pack/plugins/security_solution/cypress/tasks/api_calls/rules.ts` :)

Reply:
Thanks for this suggestion.
I've used this method as it seemed a common practice and a lot of tests using cleanKibana before starting the run.

I've tried to rewrite it with the suggestion you mentioned, but there was no performance gain. Time run was roughly the same, so I've decided to leave current implementation, as second solution looked a bit more complex